include/gbinder_fmq.h

```diff
@@ -0,0 +1,157 @@
+ *      notice, this list of conditions and the following disclaimer.
+ *   2. Redistributions in binary form must reproduce the above copyright
+ *      notice, this list of conditions and the following disclaimer in the
+ *      documentation and/or other materials provided with the distribution.
+ *   3. Neither the names of the copyright holders nor the names of its
+ *      contributors may be used to endorse or promote products derived
+ *      from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef GBINDER_FMQ_H
+#define GBINDER_FMQ_H
+
+#include <gbinder_types.h>
+
+G_BEGIN_DECLS
+
+GBinderFmq*
+gbinder_fmq_new(
+    gsize item_size,
+    gsize max_num_items,
+    gboolean configure_event_flag,
+    gint type,
+    gboolean reset_pointers,
+    gint fd,
+    gsize buffer_size);
+
+GBinderFmq*
+gbinder_fmq_ref(
+    GBinderFmq* self);
+
+void
+gbinder_fmq_unref(
+    GBinderFmq* self);
+
+/* Functions for checking how many items are available in queue */
+gsize
+gbinder_fmq_available_to_read(
+    GBinderFmq* self);
+
+gsize
+gbinder_fmq_available_to_write(
+    GBinderFmq* self);
+
+gsize
+gbinder_fmq_available_to_read_contiguous(
+    GBinderFmq* self);
+
+gsize
+gbinder_fmq_available_to_write_contiguous(
+    GBinderFmq* self);
+
+/* Functions for obtaining data pointer for zero copy read/write */
+gboolean
+gbinder_fmq_begin_read(
+    GBinderFmq* self,
+    void** data,
+    gsize items);
+
+gboolean
+gbinder_fmq_begin_write(
+    GBinderFmq* self,
+    void** data,
+    gsize items);
+
+/* Functions for ending zero copy read/write */
+gboolean
+gbinder_fmq_end_read(
+    GBinderFmq* self,
+    gsize items);
+
+gboolean
+gbinder_fmq_end_write(
+    GBinderFmq* self,
+    gsize items);
+
+/* Regular read/write functions (non-zero-copy) */
+gboolean
+gbinder_fmq_read(
+    GBinderFmq* self,
+    void* data,
+    gsize items);
+
+gboolean
+gbinder_fmq_write(
+    GBinderFmq* self,
+    const void* data,
+    gsize items);
+
+/* Blocking regular read/write functions */
+gboolean
+gbinder_fmq_read_blocking(
+    GBinderFmq* self,
+    void* data,
+    gsize items,
+    gint64 timeout_ns);
```

Comment:
Timeouts in gbinder_eventloop are expressed in ms, let's use ms here as well. (and uint for consistency)

Reply:
Changed.